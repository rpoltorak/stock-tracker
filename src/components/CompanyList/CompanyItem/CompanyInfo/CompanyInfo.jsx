import React, { Fragment } from "react";
import PropTypes from "prop-types";

CompanyInfo.propTypes = {
  domain: PropTypes.string.isRequired,
  company: PropTypes.shape({
    name: PropTypes.string,
    symbol: PropTypes.string,
    region: PropTypes.string,
    marketOpen: PropTypes.string,
    marketClose: PropTypes.string,
    timezone: PropTypes.string,
  }).isRequired,
};

export function CompanyInfo({ company, domain }) {
  return (
    <Fragment>
      <div>
        <span className="font-weight-bold mr-2">{company.name}</span>{" "}
        <span className="small mr-2">{company.symbol}</span>
        <span className="small">{domain}</span>
      </div>
      <div>
        {company.region} {company.marketOpen}-{company.marketClose}{" "}
        {company.timezone}
      </div>
    </Fragment>
  );
}
