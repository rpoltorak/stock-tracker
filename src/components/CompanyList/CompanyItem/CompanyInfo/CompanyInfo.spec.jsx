import React from "react";
import { render } from "@testing-library/react";
import { CompanyInfo } from "./CompanyInfo";

describe("CompanyInfo", () => {
  test("matches the snapshot", () => {
    const props = {
      domain: "xyz.com",
      company: {
        symbol: "GOOGL.ARG",
        name: "Alphabet Inc.",
        region: "Argentina",
        marketOpen: "11:00",
        marketClose: "17:00",
        timezone: "UTC-03",
      },
    };
    const component = render(<CompanyInfo {...props} />);

    expect(component).toMatchSnapshot();
  });
});
