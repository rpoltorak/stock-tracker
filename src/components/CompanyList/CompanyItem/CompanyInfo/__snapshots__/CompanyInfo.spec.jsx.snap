// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyInfo matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <span
          class="font-weight-bold mr-2"
        >
          Alphabet Inc.
        </span>
         
        <span
          class="small mr-2"
        >
          GOOGL.ARG
        </span>
        <span
          class="small"
        >
          xyz.com
        </span>
      </div>
      <div>
        Argentina
         
        11:00
        -
        17:00
         
        UTC-03
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <span
        class="font-weight-bold mr-2"
      >
        Alphabet Inc.
      </span>
       
      <span
        class="small mr-2"
      >
        GOOGL.ARG
      </span>
      <span
        class="small"
      >
        xyz.com
      </span>
    </div>
    <div>
      Argentina
       
      11:00
      -
      17:00
       
      UTC-03
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
